extensions/aria_extension_tosca/simple_v1_0/assignments.py

```diff
@@ -144,6 +144,17 @@ def _get_type(self, context):
             # In RelationshipAssignment
             the_type = the_type[0] # This could be a RelationshipTemplate
 
+            if isinstance(self._container._container, RequirementAssignment):
```

Comment:
alot of cascading ifs, consider using `if not isinstance(self._container._container, RequirementAssignment): return False` etc...

Reply:
+1 refactored